### PR TITLE
Revamps Cream Soda

### PR DESCRIPTION
**Pull request by EOBGames**
## About The Pull Request
Makes cream soda the antarctic treat it was always meant to be.
Changes the appearance and description of cream soda, adds cream soda bottles, and adds the pièce de résistance, the Cream Soda barrel, a violently unstable soda holder and dispenser (available from cargo for around 300 credits).

## Why It's Good For The Game
Cream soda is pretty underused as of right now, this should help revitalise it.

## Changelog

:cl:
add: The old stocks of cream soda on station have been thrown out in favour of a more purple variety. Buy a barrel from cargo today!
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -142,6 +142,39 @@
...
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	reagent_id = /datum/reagent/consumable/cream_soda
+	can_be_tanked = FALSE
+
+/obj/structure/reagent_dispensers/cream_soda/boom()
+	explosion(src, heavy_impact_range = 1, light_impact_range = 5, flame_range = 0)
+	qdel(src)
+
+/obj/structure/reagent_dispensers/cream_soda/blob_act(obj/structure/blob/B)
```
Comment (tralezab) [inline code/modules/reagents/reagent_dispenser.dm:157]: Descriptive var names, please!

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -142,6 +142,39 @@
...
+
+/obj/structure/reagent_dispensers/cream_soda/fire_act(exposed_temperature, exposed_volume)
+	boom()
+
+/obj/structure/reagent_dispensers/cream_soda/zap_act(power, zap_flags)
+	. = ..() //extend the zap
+	if(ZAP_OBJ_DAMAGE & zap_flags)
+		boom()
+
+/obj/structure/reagent_dispensers/cream_soda/bullet_act(obj/projectile/P)
```
Comment (tralezab) [inline code/modules/reagents/reagent_dispenser.dm:171]: Descriptive var names, please!

**Review by tralezab — COMMENTED** [review]
Beep boop var name time

File: code/modules/reagents/reagent_dispenser.dm (hunk 1 above)
Comment (EOBGames) [reply code/modules/reagents/reagent_dispenser.dm:157]: damn you copy paste, you've ruined me with your old variable names

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/reagent_dispenser.dm:157]: Yes

**Comment by maxymax13** [comment]
is it being purple a club penguin reference
also mmm mcodersprite ohhhhhh

**Comment by EOBGames** [comment]
> is it being purple a club penguin reference
> also mmm mcodersprite ohhhhhh

what is club penguin? I have never heard of this game, and I am certain that any purple cream soda in that game is ripping off my totally original PR here

File: code/modules/reagents/reagent_dispenser.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/reagent_dispenser.dm:157]: …/the_blob would work or something

Comment (Mothblocks) [reply code/modules/reagents/reagent_dispenser.dm:157]: or just blob

**Pull request closed without merging**